cmake/ccache.cmake

```diff
@@ -8,45 +8,57 @@ if (CMAKE_CXX_COMPILER_LAUNCHER MATCHES "ccache" OR CMAKE_C_COMPILER_LAUNCHER MA
     return()
 endif()
 
-option(ENABLE_CCACHE "Speedup re-compilations using ccache (external tool)" ON)
+option(ENABLE_CCACHE "Speedup re-compilations using ccache (default) or sccache (external tools)" ON)
+option(FORCE_SCCACHE "Use sccache by default" OFF)
 
-if (NOT ENABLE_CCACHE)
-    message(STATUS "Using ccache: no (disabled via configuration)")
+if (NOT ENABLE_CCACHE AND NOT FORCE_SCCACHE)
+    message(STATUS "Using *ccache: no (disabled via configuration)")
     return()
 endif()
 
-find_program (CCACHE_EXECUTABLE ccache)
+# We support ccache and sccache
+if (FORCE_SCCACHE)
+    find_program (CCACHE_EXECUTABLE sccache)
+else()
+    find_program (CCACHE_EXECUTABLE ccache sccache)
+endif()
+
 
 if (NOT CCACHE_EXECUTABLE)
-    message(${RECONFIGURE_MESSAGE_LEVEL} "Using ccache: no (Could not find find ccache. To significantly reduce compile times for the 2nd, 3rd, etc. build, it is highly recommended to install ccache. To suppress this message, run cmake with -DENABLE_CCACHE=0)")
+    message(${RECONFIGURE_MESSAGE_LEVEL} "Using *ccache: no (Could not find find ccache or sccache. To significantly reduce compile times for the 2nd, 3rd, etc. build, it is highly recommended to install one of them. To suppress this message, run cmake with -DENABLE_CCACHE=0)")
     return()
 endif()
 
-execute_process(COMMAND ${CCACHE_EXECUTABLE} "-V" OUTPUT_VARIABLE CCACHE_VERSION)
-string(REGEX REPLACE "ccache version ([0-9\\.]+).*" "\\1" CCACHE_VERSION ${CCACHE_VERSION})
+if (CCACHE_EXECUTABLE MATCHES "/ccache$")
```

Comment:
Yeah, that's special handling for ccache and sccache (l. 60). You might as well introduce variables `CCACHE_EXECUTABLE` and `SCCACHE_EXECUTABLE` then.

Reply:
Only one should be used in the end, so I don't see how implementing a few parameters would help.